packages/ui/src/views/repo/repo-tags/components/repo-tags-list.tsx

```diff
@@ -1,76 +1,153 @@
+    () => (tag: CommitTagType) => [
+      {
+        title: t('views:repos.createBranch', 'Create branch'),
+        onClick: () => openCreateBranchDialog(tag)
+      },
+      {
+        title: t('views:repos.viewFiles', 'View Files'),
+        to: `../code/refs/tags/${tag.name}`
+      },
+      {
+        isDanger: true,
+        title: t('views:repos.deleteTag', 'Delete tag'),
+        onClick: () => onDeleteTag(tag.name)
+      }
+    ],
+    [t, openCreateBranchDialog, onDeleteTag]
+  )
+
+  const getCreationDate = (tag: CommitTagType) => {
+    const date = new Date(tag.tagger?.when ?? 0)
+
+    return timeAgo(date.getTime())
+  }
+
+  if (!isLoading && !tagsList?.length) {
+    return (
+      <NoData
+        iconName={isDirtyList ? 'no-search-magnifying-glass' : 'no-data-tags'}
+        withBorder={isDirtyList}
+        title={isDirtyList ? t('views:noData.noResults', 'No search results') : t('views:noData.noTags', 'No tags yet')}
+        description={
+          isDirtyList
+            ? [
+                t('views:noData.checkSpelling', 'Check your spelling and filter options,'),
+                t('views:noData.changeSearch', 'or search for a different keyword.')
+              ]
+            : [
+                t(
+                  'views:noData.noTagsDescription',
+                  "Your tags will appear here once they're created. Start creating tags to see your work organized."
+                )
+              ]
+        }
+        textWrapperClassName={isDirtyList ? '' : 'max-w-[360px]'}
+        primaryButton={
+          isDirtyList
+            ? {
+                label: t('views:noData.clearSearch', 'Clear search'),
+                onClick: handleResetFiltersAndPages
+              }
+            : {
+                label: t('views:noData.createNewTag', 'Create new tag.'),
+                onClick: openCreateTagDialog
+              }
+        }
+      />
+    )
+  }
+
   return (
-    <Table.Root variant="asStackedList">
+    <Table.Root className="[&_td]:py-3.5" tableClassName="table-fixed" variant="asStackedList">
       <Table.Header>
-        <Table.Row>
-          <Table.Head className="w-2/5">{t('views:repos.tag', 'Tag')}</Table.Head>
-          <Table.Head className="w-1/5">{t('views:repos.commit', 'Commit')}</Table.Head>
-          <Table.Head className="w-1/5">{t('views:repos.date', 'Date')}</Table.Head>
-          <Table.Head className="w-1/5" />
+        <Table.Row className="pointer-events-none select-none">
+          <Table.Head className="w-[108px]">{t('views:repos.tag', 'Tag')}</Table.Head>
+          <Table.Head className="w-[330px]">{t('views:repos.description', 'Description')}</Table.Head>
+          <Table.Head className="w-[140px]">{t('views:repos.commit', 'Commit')}</Table.Head>
+          <Table.Head className="w-[146px]">{t('views:repos.tagger', 'Tagger')}</Table.Head>
+          <Table.Head className="w-[170px]">{t('views:repos.creationDate', 'Creation date')}</Table.Head>
```

Comment:
I'm not a big fan of hard-coded widths like this. If the size of the page needs to change (very likely since we need to make things fluid-width to be full-screen as is the requirement), we'll have to double back to handle that specifically here. Perhaps we can fix some widths and allow some to expand to fill the space?